src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -551,6 +559,11 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
                 util.LogErrorFromException(e, showStackTrace: true, showDetail: true, file: null);
                 ExitCode = -1;
             }
+            finally
+            {
+                _sharedCompileCts?.Dispose();
```

Comment:
Can this be null here?

Reply:
Yes, it's not the first line of the `try` hence always possible we get here via some strange exception before it's initialized.